app/Services/StandService.php

```diff
@@ -0,0 +1,131 @@
+            return [
+                'callsign' => $assignment->callsign,
+                'stand_id' => $assignment->stand_id,
+            ];
+        })->toBase();
+    }
+
+    /**
+     * Creates a stand assignment by assigning an aircraft to a stand.
+     * If the stand is already occupied, then the previous assignment is not
+     * overridden.
+     *
+     * @param string $callsign
+     * @param int $standId
+     * @throws StandAlreadyAssignedException
+     */
+    public function assignAircraftToStand(string $callsign, int $standId): void
+    {
+        $this->getStandById($standId);
+        NetworkDataService::firstOrCreateNetworkAircraft($callsign);
+        $currentAssignment = StandAssignment::where('stand_id', $standId)->first();
+
+        if ($currentAssignment && $currentAssignment->callsign !== $callsign) {
+            throw new StandAlreadyAssignedException(
+                sprintf('Stand id %d is already assigned to %s', $standId, $currentAssignment->callsign)
+            );
+        }
+
+        $assignment = StandAssignment::updateOrCreate(
+            ['callsign' => $callsign],
+            [
+                'callsign' => $callsign,
+                'stand_id' => $standId,
+            ]
+        );
+
+        event(new StandAssignedEvent($assignment));
+    }
+
+    /**
+     * Creates a stand assignment by assigning a particular stand to an aircraft.
+     * If the stand is already assigned, that assignment is first removed.
+     *
+     * @param string $callsign
+     * @param int $standId
+     */
+    public function assignStandToAircraft(string $callsign, int $standId): void
+    {
+        $this->getStandById($standId);
+
+        NetworkDataService::firstOrCreateNetworkAircraft($callsign);
+        $currentAssignment = StandAssignment::with('aircraft')
+            ->where('stand_id', $standId)
+            ->first();
+
+        if ($currentAssignment && $currentAssignment->callsign !== $callsign) {
+            $this->deleteStandAssignment($currentAssignment->aircraft->callsign);
+        }
+
+        $assignment = StandAssignment::updateOrCreate(
+            ['callsign' => $callsign],
+            [
+                'stand_id' => $standId,
+            ]
+        );
+
+        event(new StandAssignedEvent($assignment));
+    }
+
+    public function deleteStandAssignment(string $callsign): void
+    {
+        if (!StandAssignment::destroy($callsign)) {
+            return;
+        }
+
+        event(new StandUnassignedEvent($callsign));
+    }
+
+    public function getDepartureStandAssignmentForAircraft(NetworkAircraft $aircraft): ?StandAssignment
+    {
+        return StandAssignment::where('callsign', $aircraft->callsign)
+            ->whereHas('stand.airfield', function (Builder $query) use ($aircraft) {
+                $query->where('code', $aircraft->planned_depairport);
+            })
+            ->first();
+    }
+
+    private function getStandById(int $standId): Stand
```

Comment:
I feel like because you're not assigning this anywhere, it just needs to check if the site is found and thus the name isn't ideal, and then throw the exception.

Reply:
Changed method to standExists